syntax = "proto3";

package rag ;

// The gRPC service definition.
service RagService {
// Processes a document: reads, chunks, embeds, and stores it.
rpc ProcessDocument(ProcessRequest) returns (ProcessResponse) {}
// Answers a query based on the document context.
rpc AnswerQuery(QueryRequest) returns (QueryResponse) {}
}

// The request message for processing a document.
message ProcessRequest {
string document_path = 1                          ;
string chat_id = 2                                ;
}

// The response message after processing a document.
message ProcessResponse {
bool success = 1                                     ;
string message = 2                                   ;
}

// The request message for answering a query.
message QueryRequest {
string query_text = 1                         ;
string chat_id = 2                            ;
}

// The response message containing the AI's answer.
message QueryResponse {
string answer = 1                                   ;
}
